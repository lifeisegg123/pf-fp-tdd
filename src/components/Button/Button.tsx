import { MouseEventHandler, ReactNode } from "react";
import { ComponentBaseProps } from "src/interfaces";

interface ButtonProps extends ComponentBaseProps {
  onClick?: MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
  children?: ReactNode;
}

export function Button(props: ButtonProps) {
  return <button {...props} />;
}
