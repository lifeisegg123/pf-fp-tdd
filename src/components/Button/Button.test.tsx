import { Button } from "./Button";
import { render, fireEvent } from "@testing-library/react";

describe("Component/Button", () => {
  it("should match snapshot", () => {
    const { container } = render(<Button>test</Button>);
    expect(container).toMatchSnapshot();
  });

  it("should handle click event", () => {
    const mockFn = jest.fn();
    const { getByRole } = render(<Button onClick={mockFn} />);
    const button = getByRole("button") as HTMLButtonElement;
    fireEvent.click(button);
    expect(mockFn).toBeCalled();
  });

  it("should be disabled", () => {
    const mockFn = jest.fn();
    const { getByRole } = render(<Button disabled onClick={mockFn} />);

    const button = getByRole("button") as HTMLButtonElement;

    fireEvent.click(button);

    expect(button.disabled).toEqual(true);
    expect(mockFn).not.toBeCalled();
  });
});
