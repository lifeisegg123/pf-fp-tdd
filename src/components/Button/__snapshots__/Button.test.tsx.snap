// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/Button should match snapshot 1`] = `
<div>
  <button>
    test
  </button>
</div>
`;
